### PR TITLE
Add bz2 extension to bz2 files before attempting to extract. Fixes #105

### LINKED ISSUE #105: 📦 ziextract should rename .bzip2 files before attempting to extract
While working on https://github.com/zdharma-continuum/zinit-packages/pull/2 I noticed that `ziextract` does detect bzip2 files correctly but ultimately fails to extract them when their suffix does *not* match `bz2|bzip2`.

This is actually a `bunzip2` issue, it refuses to operate on files whose extension don't match.

```
bunzip2: zinit-package-apr: unknown suffix - ignored
```

This is easily fixable by renaming the file just before attempting the extraction. The same is already done for zx packages.

Refs:
- See the `apr` package: https://github.com/zdharma-continuum/zinit-packages/blob/main/apr/package.json

### PR DESCRIPTION
**Pull request by pschmitt**

### CODE REVIEW

File: zinit-install.zsh [hunk 1]
```diff
@@ -1683,6 +1683,13 @@ ziextract() {
...
             ;;
         ((#i)*.bz2|(#i)*.bzip2)
+            # Rename file if its extension does not match "bz2". bunzip2 refuses
+            # to operate on files that are not named correctly.
+            # See https://github.com/zdharma-continuum/zinit/issues/105
+            if [[ $file != (#i)*.bz2 ]] {
+                command mv $file $file.bz2
+                file=$file.bz2
+            }
             →zinit-extract() { →zinit-check bunzip2 "$file" || return 1
```
Comment (alichtman) [inline zinit-install.zsh:1686]: Looks like a missed Unicode char

**Review by alichtman — APPROVED** [review]

File: zinit-install.zsh (hunk 1 above)
Comment (pschmitt) [reply zinit-install.zsh:1686]: "missed" by whom?
This is a function name!

Comment (pschmitt) [reply zinit-install.zsh:1686]: https://github.com/zdharma-continuum/zinit/issues/58 only covered file names

Comment (alichtman) [reply zinit-install.zsh:1686]: Ah, gotcha. We should spin off a task for removing those from function names. I don't think we should have any unicode chars in the codebase except for strings we're printing for the user. It makes the code a lot harder to work with

Comment (alichtman) [reply zinit-install.zsh:1686]: https://github.com/zdharma-continuum/zinit/issues/108

**Pull request merged**